baseplate.go

```diff
@@ -77,17 +77,40 @@ type Server interface {
 	Serve() error
 }
 
-// Serve runs the given Server until it is given an external shutdown signal
-// using runtimebp.HandleShutdown to handle the signal and shut down the
-// server gracefully.  Returns the (possibly nil) error returned by "Close" or
+// ServeArgs provides a list of arguments to Serve.
+type ServeArgs struct {
+	// Server: Mandatory. Indicates the Server that should be run until receiving
+	// a shutdown signal.
+	Server Server
+
+	// PreShutdown: Optional. Indicates a slice of io.Closers that should be
+	// gracefully shut down before the server upon receipt of a shutdown signal.
+	PreShutdown []io.Closer
+
+	// PostShutdown: Optional. Indicates a slice of io.Closers that should be
+	// gracefully shut down after the server upon receipt of a shutdown signal.
+	PostShutdown []io.Closer
+}
+
+// Serve runs the given Server until it is given an external shutdown signal.
+//
+// It uses runtimebp.HandleShutdown to handle the signal and gracefully shut
+// down, in order:
+//     * any provided PreShutdown closers,
+//     * the Server, and
+//     * any provided PostShutdown closers.
```

Comment:
super nit: this way the 3 bullet points will be rendered as code block in the godoc site, which works but might look weird. it's slightly better to not put 4 extra spaces before them, and add blank lines between them, e.g.:

```
// other comments
//
// * bullet 1
//
// * bullet 2
```

Reply:
Great nit, and thanks for pointing me to the name `godoc` so I know what to google for now :P